src/runtime/server/lib/oauth/microsoft.ts

```diff
@@ -168,22 +169,61 @@ export function microsoftEventHandler({ config, onSuccess, onError }: OAuthConfi
       }
     }
     else {
-      // use of any is required as MS has two token versions
-      // see: https://learn.microsoft.com/en-us/entra/identity-platform/access-tokens
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      const decodedToken = jwtDecode<any>(accessToken)
-      const msJwtVersion: '1.0' | '2.0' = decodedToken.ver
-
-      if (msJwtVersion === '2.0') {
-        user.displayName = decodedToken.name
-        user.mail = decodedToken.preferred_username
+      // required to unsafely decode to get the Kid from the header
+      const decoded = jwt.decode(accessToken, { complete: true })
+      if (!decoded) {
+        const error = createError({
+          statusCode: 401,
+          message: `Microsoft login failed: ${user.error || 'Failed to decoded JWT'}`,
+        })
+        if (!onError) throw error
+        return onError(event, error)
       }
-      else {
-        const firstName = decodedToken.given_name
-        const lastName = decodedToken.family_name
-        user.displayName = `${firstName} ${lastName}`
-        user.mail = decodedToken.unique_name
+
+      const kid = decoded.header.kid
+      if (!kid) {
+        const error = createError({
+          statusCode: 401,
+          message: `Microsoft login failed: ${user.error || 'Missing Kid'}`,
+        })
+        if (!onError) throw error
+        return onError(event, error)
       }
+
+      const client = jwksClient({
+        jwksUri: 'https://login.microsoftonline.com/common/discovery/keys',
```

Comment:
I would keep it configurable

Reply:
Updated to be part of the config

Reply:
@amandesai01 just nudging you on this when you get a chance. I deployed with my forked version on a private npm registry but would like to switch back to the official repo once merged.